### PR TITLE
test: add test cases for configuration of daemon

### PR DESCRIPTION
**Pull request by chuanchang**
Signed-off-by: Alex Jia <chuanchang.jia@gmail.com>

### CODE REVIEW

File: daemon/config/config.go [hunk 1]
```diff
@@ -166,6 +166,7 @@ func (cfg *Config) MergeConfigurations(flagSet *pflag.FlagSet) error {
 	contents, err := ioutil.ReadFile(cfg.ConfigFile)
 	if err != nil {
 		if os.IsNotExist(err) {
+			logrus.Debugf("the %v doesn't exist: %v", cfg.ConfigFile, err)
```
Comment (ZYecho) [inline daemon/config/config.go]: more details? such as the config file  %v doesn't exist?

File: daemon/config/config_test.go [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+	"io/ioutil"
+	"os"
+	"strings"
 	"testing"
 
+	"github.com/alibaba/pouch/apis/types"
+	"github.com/alibaba/pouch/client"
+	criconfig "github.com/alibaba/pouch/cri/config"
+	"github.com/alibaba/pouch/network"
+	"github.com/alibaba/pouch/storage/volume"
```
Comment (ZYecho) [inline daemon/config/config_test.go]: add blank line below this line.

File: daemon/config/config_test.go [hunk 3]
```diff
@@ -48,7 +58,170 @@ func TestFlattenConfig(t *testing.T) {
...
+	assert.Equal(nil, cfg.Validate())
+
+	cfg = &Config{
+		Labels: []string{
+			"foo=bar",
+		},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	// Test others configuration
```
Comment (ZYecho) [inline daemon/config/config_test.go]: Does this test need to cover invalid configuration case?

File: daemon/config/config.go (hunk 1 above)
Comment (chuanchang) [reply daemon/config/config.go]: I think it should be enough to write a log in here.

File: daemon/config/config_test.go (hunk 2 above)
Comment (chuanchang) [reply daemon/config/config_test.go]: sure.

File: daemon/config/config_test.go (hunk 3 above)
Comment (chuanchang) [reply daemon/config/config_test.go]: Absolutely yes,  unit test need to cover all of code path, including negative path.

**Comment by chuanchang** [comment]
the v2 patches will fix CI failure and  update according to above advice.

**Comment by allencloud** [comment]
I am afraid that the CI is not happy 😢 
```
#!/bin/bash -eo pipefail
gometalinter --disable-all --skip vendor -E gofmt -E goimports -E golint -E goconst -E ineffassign -E misspell -E vet -E megacheck ./...
daemon/config/config_test.go:1::warning: file is not gofmted with -s (gofmt)
daemon/config/config_test.go:1::warning: file is not goimported (goimports)
Exited with code 1
```

Could you check the gofmt and goimports thing? @chuanchang

File: daemon/config/config_test.go [hunk 4]
```diff
@@ -80,3 +252,54 @@ func TestGetConflictConfigurations(t *testing.T) {
...
+	// Test invalid content format
+	configFile := "/tmp/test-merge-config-1.json"
+	err := ioutil.WriteFile(configFile, []byte(""), 0644)
+	assert.Equal(nil, err)
+
+	cfg = &Config{
+		ConfigFile: configFile,
+	}
+
+        err = cfg.MergeConfigurations(flags)
```
Comment (allencloud) [inline daemon/config/config_test.go]: I think you do not the `gofmt` this place?
I did not check everywhere, while I think it is easy to use `gomft` to format all the file. @chuanchang

Comment (chuanchang) [reply daemon/config/config_test.go]: sure, I did gofmt for v1 patch, but forgot it for v2, thanks for pointing it out.

File: daemon/config/config_test.go [hunk 5]
```diff
@@ -48,7 +59,168 @@ func TestFlattenConfig(t *testing.T) {
...
+			RemoveVolume:   false,
+			DefaultBackend: "/path/to",
+			VolumeMetaPath: "/foo/bar",
+			DriverAlias:    "local=alilocal",
+		},
+	}
+	assert.Equal(nil, cfg.Validate())
+
+	// Test network configuration
+	cfg = &Config{
```
Comment (fuweid) [inline daemon/config/config_test.go]: @chuanchang The config validate only checks the labels and duplicate listeners. If we don't add more checker, the test case will be no-op here. Could you help to do that?

https://github.com/alibaba/pouch/blob/master/daemon/config/config.go#L130-L162

Comment (chuanchang) [reply daemon/config/config_test.go]: sure, I know this part code, but I lost this scenario when wrote testing, the next version will cover it soon.

**Comment by allencloud** [comment]
LGTM

**Pull request merged**